src/azure-cli-core/azure/cli/core/tests/test_keys.py

```diff
@@ -0,0 +1,110 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+import unittest
+import tempfile
+import paramiko
+import io
+import os
+import logging
+
+from ..keys import generate_ssh_keys
+
+
+class TestGenerateSSHKeys(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        # set up temporary directory to be used for temp files.
+        cls._tempdir = tempfile.TemporaryDirectory()
+        cls._tempdirName = cls._tempdir.name
+
+        cls.key = paramiko.RSAKey.generate(2048)
+
+        keyOutput = io.StringIO()
+        cls.key.write_private_key(keyOutput)
+
+        cls.private_key = keyOutput.getvalue()
+        cls.public_key = '%s %s' % (cls.key.get_name(), cls.key.get_base64())
+
+    @classmethod
+    def tearDownClass(cls):
+        # clean up temporary directory to be used for temp files.
+        cls._tempdir.cleanup()
+
+    def test_public_key_file_exists(self):
+
+        # Create public key file
+        public_key_path = self._create_new_temp_key_file(self.public_key, suffix=".pub")
+
+        # Call generate_ssh_keys and assert that returned public key same as original
+        new_public_key = generate_ssh_keys("", public_key_path)
+        self.assertEqual(self.public_key, new_public_key)
+
```

Comment:
Might also want to check that private key is safe.

Reply:
Okay, will do

Reply:
Also I checked the CI, but wasn't sure what the bug was, I re-ran it and would probably ask for clarification if I don't understand it.